Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py

```diff
@@ -889,11 +911,61 @@ def __init__(self, id_to_object: dict[str, Any], verify: bool = True,
             re.compile(CIDR_ISSUBSET_VAL_PATTERN),
             re.compile(CIDR_ISUPPERSET_VAL_PATTERN),
         ]
-        self.field_map = field_map if field_map else {}
+        self.field_map = field_map or {}
         self.update_custom_fields = update_custom_fields
-        self.tags = tags if tags else []
+        self.tags = tags or []
         self.last_fetched_indicator__modified = None
 
+    @staticmethod
+    def get_pattern_comparisons(pattern: str) -> Optional[PatternComparisons]:
+        """
+        Parses a pattern and comparison and extracts the comparisons as a dictionary.
+        If the pattern is invalid, the return value will be "None".
+
+        For Example:
+
+        >>> STIX2XSOARParser.get_pattern_comparisons(
+        >>>     "[ipv4-addr:value = '1.51.100.1/32' "
+        >>>     "OR ipv4-addr:value = '203.0.113.33/32' "
+        >>>     "AND domain-name:value = 'example.com' "
+        >>>     "OR file:hashes.'SHA-256' = '13987239847...']"
+        >>> )
+        {
+            'ipv4-addr': [(['value'], '=', "'1.51.100.1/32'"), (['value'], '=', "'203.0.113.33/32'")],
+            'domain-name': [(['value'], '=', "'example.com'")],
+            'file': [(['hashes', 'SHA-256'], '=', "'13987239847...'")]
+        }
+
+        Args:
+            pattern: the pattern to extract the value from.
+
+        Returns:
+            Optional[PatternComparisons]. the value in the pattern.
+        """
+        try:
+            return cast(PatternComparisons, Pattern(pattern).inspect().comparisons)
+        except Exception as error:
+            demisto.debug(f'Unable to parse {pattern=}, {error=}')
```

Comment:
add indicator id to the log

Reply:
The indicator ID is logged when the indicator is skipped.